src/ConnectionPool.jl

```diff
@@ -270,13 +290,20 @@ end
 Wait for prior pending reads to complete.
 """
 function IOExtras.startread(t::Transaction)
-    @require !isreadable(t)                      ;t.c.readcount != t.sequence &&
-                                                   @debug 1 "⏳  Wait read:  $t"
+    @require !isreadable(t)
+
+    readbusy(t.c) && @debug 1 "⏳  Wait read:  $t"
     t.c.timestamp = time()
-    while t.c.readcount != t.sequence
-        wait(t.c.readdone)
-    end                                           ;@debug 2 "👁  Start read: $t"
-    t.c.readbusy = true
+    while true
```

Comment:
busy waiting on a condition variable is generally a pretty strong anti-pattern

Reply:
Yes, this was one of the last changes I made because I saw some test cases where pipelined requests got out of order; e.g. you have multiple pipelined requests and when one finishes writing, it needs to be the next transaction to read from that same connection; otherwise you get requests that end up with responses to other requests on the same connection.

This code seems to solve it, though it definitely felt "icky".

Reply:
Moving this back to a `lock` -> check -> `wait` pattern on a `Condition` feels much more standard.